components/ReduceOneItem.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from 'react';
+import cookie from 'js-cookie';
+
+type Item = {
+  id: string;
+  name: string;
+  img: string;
+  amount: number;
+  price: number;
+  info: string;
+};
+
+type Cart = {
+  id: string;
+  name: string;
+  img: string;
+  amount: number;
+  price: number;
+  info: string;
+};
+
+type Products = {
+  id: string;
+  name: string;
+  img: string;
+  amount: number;
+  price: number;
+  info: string;
+};
+type Props = { item: Item; cart: Cart[]; products: Products[] };
+export default function ReduceOneItem(props: Props) {
+  // const [itemsInCart, setItemsInCart] = useState(props.itemsInCart || []);
+
+  // useEffect(() => {
+  //   cookie.set('cart', props.itemsInCart);
+  // }, [props.itemsInCart]);
+
+  function reduceOne(id) {
+    const newCart = props.cart.map((item) => {
+      if (item.id === id) {
+        if (item.amount === 1) {
+          alert('Use remove button!');
```

Comment:
Maybe a better UX would be to not even show the button at all if the `props.item.amount` is `1`?